feature-libs/checkout/base/components/components/delivery-mode/delivery-mode.component.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { CheckoutDeliveryModesFacade } from '@spartacus/checkout/base/root';
+import { DeliveryMode, FeatureConfigService } from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { CheckoutConfigService } from '../../services/checkout-config.service';
+import { CheckoutStepService } from '../../services/checkout-step.service';
+
+@Component({
+  selector: 'cx-delivery-mode',
+  templateUrl: './delivery-mode.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DeliveryModeComponent implements OnInit, OnDestroy {
+  supportedDeliveryModes$: Observable<DeliveryMode[]>;
+  selectedDeliveryMode$: Observable<DeliveryMode>;
+  continueButtonPressed = false;
+
+  backBtnText = this.checkoutStepService.getBackBntText(this.activatedRoute);
+
+  deliveryModeSub: Subscription;
+
+  mode: FormGroup = this.fb.group({
+    deliveryModeId: ['', Validators.required],
+  });
+
+  constructor(
+    private fb: FormBuilder,
+    private checkoutConfigService: CheckoutConfigService,
+    private activatedRoute: ActivatedRoute,
+    protected checkoutStepService: CheckoutStepService,
+    protected checkoutDeliveryModesService: CheckoutDeliveryModesFacade,
+    protected featureConfigService: FeatureConfigService
+  ) {}
+
+  ngOnInit(): void {
+    this.supportedDeliveryModes$ = this.checkoutDeliveryModesService
+      .getSupportedDeliveryModesState()
```

Comment:
Should we keep this function to return a queryState instead of the actual data only?

For example, since we're only using the data, let's have the service return the data instead of the state. What do you think? Or create another function that returns the actual data and leave the state function in the service for future use.

Asking because, we have an instance where we do both like in `checkout-payment.service.ts` in regards to `getCardtypes*`

Reply:
Added a method that returns the data directly, or an empty array. I'd like to _always_ expose the _*state_ methods, and _sometimes_ introduced a shortcut function which returns the data directly - usually, when we need it ourselves. I'd like to avoid introducing methods just for the sake of it, until we are confident in our approach